infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -200,17 +199,16 @@ private void createStreamNameAndVersionTables(Set<String> streams) {
             CORFU_SYSTEM_NAMESPACE, TableRegistry.PROTOBUF_DESCRIPTOR_TABLE_NAME);
         streams.add(protoTable);
 
-        LogReplicationStreams.VersionString versionString = VersionString.newBuilder()
-            .setName("VERSION").build();
+        VersionString versionString = VersionString.newBuilder()
+            .setName(VERSION_PLUGIN_KEY).build();
+        currentVersion = logReplicationConfigAdapter.getVersion();
         LogReplicationStreams.Version version = Version.newBuilder()
-            .setVersion(logReplicationConfigAdapter.getVersion()).build();
+            .setVersion(currentVersion).build();
```

Comment:
shouldn't you be setting the upgrade flag here? missed seeing where it is set..

Reply:
Done. I was thinking to left that field with its `false` default value, but yeah it makes more sense to explicitly set it here.